File: electro.client/types/Api/PagedResult.ts
export interface IPagedResult<TResult> extends IPaginationResult {
	data: TResult[];
}

export interface IPaginationParams {
	pageNumber: number;
	pageSize: number;
}

export interface IPaginationResult {
	pageNumber: number;
	pageSize: number;
	totalItems: number;
	totalPages: number;
}

File: electro.api.rest/Utils/PagedResult/PagedResult.cs

namespace electro.api.rest.Utils.PagedResult
{
    public class PagedResult<T>: PaginationInfo
    {
        public IEnumerable<T> Data { get; set; }
        public PagedResult(IEnumerable<T> items, int pageNumber, int pageSize, int totalItems)
        {
            Data = items;
            PageNumber = pageNumber;
            PageSize = pageSize;
            TotalItems = totalItems;
            TotalPages = (int)Math.Ceiling(totalItems / (double)pageSize);
        }
    }
}
